In [2]:
import pandas as pd

# Load the dataset
df = pd.read_csv('host_type_all_preds.csv')

# View the first 5 rows
df.head()

,proteinID,host_type_lysin,host_type_endolysin,host_type_val,host_type_capsid,host_type_baseplate,host_type_tail_appendage,host_type_tail_sheath,host_type_portal,host_type_head-tail_joining,...,host_type_sir2,host_type_toxin,host_type_super_infection,host_type_lysis,host_type_lysis_inhibitor,host_type_holin,host_type_spanin,host_type_tail,host_type_pvp,host_type_DNA-associated
0,AB002632_00001,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.003104
1,AB002632_00002,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.139036
2,AB002632_00003,NaN,NaN,NaN,0.029791,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.793287,NaN
3,AB002632_00004,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,AB002632_00005,NaN,NaN,NaN,0.008570,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.015858,NaN


a prediction score from one of the ~40 function-specific models for a given host type (gram-positive or gram-negative — more on this below). So host_type_capsid = 0.029791 means "the capsid-function model gives this protein a 2.9% probability of being associated with this host type."